firmware_motion_board/src/control_system.c

```diff
@@ -0,0 +1,135 @@
+#include "control_system.h"
+
+volatile PidReg_t reg_left = PIDREG_DEFAULTS;
+volatile PidReg_t reg_right = PIDREG_DEFAULTS;
+
+void control_system_initRegulators()
+{    
+    reg_left.clamp_min = -(float)MOTOR_MAX_PWM;
+    reg_left.clamp_max = +(float)MOTOR_MAX_PWM;
+    
+    reg_left.ref = 0;
+    reg_left.out = 0;
+    reg_left.integrator = 0;
+    
+    reg_left.kp = 120.0;
+    reg_left.ki = 1.5;
+    reg_left.kd = 0.0;
+    
+    reg_right.ref = 0;
+    reg_right.out = 0;
+    reg_right.integrator = 0;
+    
+    reg_right.kp = 120.0;
+    reg_right.ki = 1.5;
+    reg_right.kd = 0.0;
+    
+    reg_right.clamp_min = -(float)MOTOR_MAX_PWM;
+    reg_right.clamp_max = +(float)MOTOR_MAX_PWM;
+}
+
+void control_system_resetRegulators()
+{
+    PidReg_reset(&reg_left);
+    PidReg_reset(&reg_right);
+}
+
+/*Call this at fixed frequency!!!*/
+void control_system_interrupt()
+{
+    int32_t v_left, v_right;
+    v_left = -(int32_t)QEI3_VelocityGet();
+    v_right = (int32_t)QEI1_VelocityGet();
+    
+    reg_left.fdb = (float)v_left;
+    reg_right.fdb = (float)v_right;

```

Comment:
Zasto ga ovdje kastujemo, zasto ga ne deklarisemo kao int32_t?

Reply:
Nema vise kastovanja, sad je to int32_t. (izmenjeno je)